packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormat.ts

```diff
@@ -1,25 +1,61 @@
-import type { W3cCredential } from '../../../vc/models/credential/W3cCredential'
+import type { JsonObject } from '../../../../types'
+import type { SingleOrArray } from '../../../../utils'
+import type { LinkedDataProof } from '../../../vc'
+import type { IssuerOptions } from '../../../vc/models/credential/Issuer'
 import type { CredentialFormat } from '../CredentialFormat'
 import type { JsonLdOptionsRFC0593 } from './JsonLdOptionsRFC0593'
 
-export interface JsonLdAcceptRequestOptions extends SignCredentialOptionsRFC0593 {
-  verificationMethod?: string
+export interface JsonCredential {
+  '@context': Array<string> | JsonObject
+  id?: string
+  type: Array<string>
+  issuer: string | IssuerOptions
+  issuanceDate: string
+  expirationDate?: string
+  credentialSubject: SingleOrArray<JsonObject>
+  [key: string]: unknown
+}
+
+// this is the API interface (only)
+export interface JsonLdSignCredentialFormat {
+  credential: JsonCredential
+  options: JsonLdOptionsRFC0593
 }
 
+// use this interface internally as the above may diverge in future
 export interface SignCredentialOptionsRFC0593 {
-  credential: W3cCredential
+  credential: JsonCredential
   options: JsonLdOptionsRFC0593
 }
 
+export interface JsonVerifiableCredential extends JsonLdSignCredentialFormat {
+  proof: LinkedDataProof
```

Comment:
LinkedDataProof is a class, so it won't have the right interface. We should create a custom interface that is just for the json representation of the jsonld credential. See my comment here of what the interface could look like: https://github.com/hyperledger/aries-framework-javascript/pull/1140#discussion_r1045566257